### PR TITLE
Update test structure and code to make integration tests easier to work with

### PR DESCRIPTION
**Pull request by leandrocp**
- Introduce fixtures
- Introduce ConnCase to build a conn
- DummyApp -> Beacon.BeaconTest
- DRY endpoint config
- Keep a pattern for creating test data (fixtures)
- Remove unnecessary code

### CODE REVIEW

File: test/support/fixtures.ex [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  alias Beacon.Components
+  alias Beacon.Layouts
+  alias Beacon.Pages
+  alias Beacon.Stylesheets
+
+  def get_lazy(attrs, key, fun) when is_map(attrs), do: Map.get_lazy(attrs, key, fun)
+  def get_lazy(attrs, key, fun), do: Keyword.get_lazy(attrs, key, fun)
+
+  def stylesheet_fixture(attrs \\ %{}) do
+    Enum.into(attrs, %{
```
Comment (KronicDeth) [inline test/support/fixtures.ex:11]:
```suggestion
    attrs
    |> Enum.into(%{
```

File: test/support/fixtures.ex [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      site: "my_site",
+      name: "sample_stylesheet",
+      content: "body {cursor: zoom-in;}"
+    })
+    |> Stylesheets.create_stylesheet!()
+  end
+
+  def component_fixture(attrs \\ %{}) do
+    Enum.into(attrs, %{
```
Comment (KronicDeth) [inline test/support/fixtures.ex:20]:
```suggestion
    attrs
    |> Enum.into(%{
```

File: test/support/fixtures.ex [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+      """
+    })
+    |> Layouts.create_layout!()
+  end
+
+  def page_fixture(attrs \\ %{}) do
+    layout_id = get_lazy(attrs, :layout_id, fn -> layout_fixture().id end)
+
+    Enum.into(
+      attrs,
```
Comment (KronicDeth) [inline test/support/fixtures.ex:49]:
```suggestion
    attrs
    |> Enum.into(
```

File: test/support/fixtures.ex [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        """
+      }
+    )
+    |> Pages.create_page!()
+  end
+
+  def page_event_fixture(attrs \\ %{}) do
+    page_id = get_lazy(attrs, :page_id, fn -> page_fixture().id end)
+
+    Enum.into(attrs, %{
```
Comment (KronicDeth) [inline test/support/fixtures.ex:67]:
```suggestion
    attrs
    |> Enum.into(%{
```

File: test/support/fixtures.ex [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        {:noreply, assign(socket, :message, "Hello \#{event_params["greeting"]["name"]}!")}
+      """
+    })
+    |> Pages.create_page_event!()
+  end
+
+  def page_helper_fixture(attrs \\ %{}) do
+    page_id = get_lazy(attrs, :page_id, fn -> page_fixture().id end)
+
+    Enum.into(attrs, %{
```
Comment (KronicDeth) [inline test/support/fixtures.ex:80]:
```suggestion
    attrs
    |> Enum.into(%{
```

**Review by KronicDeth — COMMENTED** [review]
Start pipelines using `Enum` with the enumerable.

**Comment by AZholtkevych** [comment]
Thanks a lot guys, @KronicDeth @leandrocp

**Pull request merged**